MockNgRedux with incompatible types

### This is a...

* [ ] feature request
* [x] bug report
* [ ] usage question

### What toolchain are you using for transpilation/bundling?

* [x] `@angular/cli`
* [ ] Custom @ngTools/webpack
* [ ] Raw `ngc`
* [ ] SystemJS
* [ ] Rollup
* [ ] Other

### Environment

NodeJS Version:  8.9.1
Typescript Version: 2.6.2
Angular Version: 5.1.3
@angular-redux/store version: 7.1.0
@angular/cli version: 1.6.4
karma: 2.0.0

OS: Windows 10 64 bits

### Expected Behaviour:
To be able to execute a test for my DemoComponent using Karma and MockNgRedux
```
import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { ReactiveFormsModule } from '@angular/forms';
import {
  MatCardModule,
  MatSelectModule
} from '@angular/material';

import { MockNgRedux, NgReduxTestingModule } from '@angular-redux/store/testing';

import { DemoComponent } from './demo.component';


describe('DemoComponent', () => {
  let component: DemoComponent;
  let fixture: ComponentFixture<DemoComponent>;

  beforeEach(async(() => {
    void TestBed.configureTestingModule({
      imports: [
        MatCardModule,
        MatSelectModule,
        NgReduxTestingModule,
        ReactiveFormsModule
      ],
      declarations: [
        DemoComponent
      ]
    })
    .compileComponents();

    MockNgRedux.reset();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(DemoComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
});
```


### Actual Behaviour:
As soon as I add the line `MockNgRedux.reset();`, I get a compilation error when running the karma test
`npm run test`


### Stack Trace/Error Message:
```
 10% building modules 1/1 modules 0 active25 01 2018 09:55:42.926:INFO [karma]: Front-end scripts not present. Compiling...                                             25 01 2018 09:55:43.110:ERROR [karma]: Error: node_modules/@angular-redux/store/lib/testing/ng-redux.mock.d.ts(12,22): error TS2415: Class 'MockNgRedux<T>' incorrectly extends base class 'NgRedux<T>'.
  Types of property 'select' are incompatible.
    Type '<SelectedType>(selector: Selector<T, SelectedType>, comparator?: Comparator | undefined) => Obser...' is not assignable to type '<SelectedType>(selector?: string | number | symbol | (string | number)[] | ((s: T) => SelectedTyp...'.
      Types of parameters 'selector' and 'selector' are incompatible.
        Type 'string | number | symbol | (string | number)[] | ((s: T) => SelectedType) | undefined' is not assignable to type 'Selector<T, SelectedType>'.
          Type 'undefined' is not assignable to type 'Selector<T, SelectedType>'.
```

--- a/testing/ng-redux.mock.ts
+++ b/testing/ng-redux.mock.ts
@@ -83,7 +83,7 @@ export class MockNgRedux<T = {}> extends NgRedux<T> {
   configureSubStore = this.mockRootStore.configureSubStore;
 
   select: <SelectedType>(
-    selector: Selector<T, SelectedType>,
+    selector?: Selector<T, SelectedType>,
     comparator?: Comparator
   ) => Observable<SelectedType> = this.mockRootStore.select;
 

